### PR TITLE
Feature: Add a select dropdown on create challenge page

### PR DESCRIPTION
**Pull request by Mustafa-Esoofally**
image

### CODE REVIEW

File: src/app/admin/TemplateManager.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    name: string;
+    content: string;
+}
+
+export function TemplateManager() {
+    const [templates, setTemplates] = useState<Template[]>([]);
+    const [isModalOpen, setIsModalOpen] = useState(true);
+    const [currentTemplate, setCurrentTemplate] = useState<Template | null>(null);
+
+    useEffect(() => {
```
Comment (noncuro) [inline src/app/admin/TemplateManager.tsx:16]: useQuery

File: src/app/admin/TemplateManager.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        fetchTemplates();
+    }, []);
+
+    const fetchTemplates = async () => {
```
Comment (noncuro) [inline src/app/admin/TemplateManager.tsx:20]: Would extract from this function to top-level

File: src/app/admin/createChallenge.tsx [hunk 3]
```diff
@@ -1,36 +1,182 @@
...
     const [token, setToken] = useState('');
+    const [templates, setTemplates] = useState<Template[]>([]);
+    const [selectedTemplate, setSelectedTemplate] = useState<Template | null>(null);
+    const [isModalOpen, setIsModalOpen] = useState(false);
+
+    useEffect(() => {
+        fetchTemplates();
+    }, []);
+
+    const fetchTemplates = async () => {
```
Comment (noncuro) [inline src/app/admin/createChallenge.tsx:79]: todo: use query

**Review by noncuro — APPROVED** [review]
Just leave todos and good with me :)

**Comment by Mustafa-Esoofally** [comment]
@noncuro Made the changes as requested

**Comment by Mustafa-Esoofally** [comment]
https://github.com/noncuro/challenge-platform/pull/6 addresses all the changed needed

**Pull request merged**